src/twisted/mail/protocols.py

```diff
@@ -37,7 +38,7 @@ class DomainDeliveryBase:
         Sub-classes should set this appropriately.
     """
     service = None
-    protocolName = None
+    protocolName = None  # type: Optional[bytes]
```

Comment:
I don't think that this is optional. protocolName should always be defined, as described by the docstring.

Maybe have something like this.
```suggestion
    protocolName = b'not-implemented-protocol`
```

Reply:
Done